src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -746,15 +756,15 @@ public void Dispose()
             }
         }
 
-        private unsafe void ReadItemsFromSource(int replicatedDocs, DocumentsOperationContext context,
+        private unsafe List<ReplicationItem> ReadItemsFromSource(int replicatedDocs, DocumentsOperationContext context,
             IncomingReplicationAllocator incomingReplicationAllocator, IncomingReplicationStatsScope stats)
         {
+            var replicatedItems = new List<ReplicationItem>();
```

Comment:
Can we avoid this allocation? Return `Span<ReplicationItem>` and user `ArrayPool`, instead?

Reply:
done